source/MaterialXRuntime/RtApi.cpp

```diff
@@ -274,13 +279,16 @@ class PvtApi
         _libraryRoot.reset();
         _libraries.clear();
         _libraryRoot = RtStage::createNew(libRootName);
+
+        _unitDefinitions = UnitConverterRegistry::create();
     }
 
     FileSearchPath _searchPaths;
     FileSearchPath _implementationSearchPaths;
     FileSearchPath _textureSearchPaths;
     RtStagePtr _libraryRoot;
     RtTokenMap<RtStagePtr> _libraries;
+    UnitConverterRegistryPtr  _unitDefinitions;
```

Comment:
Smallest non document construct we have is a unit conversion registry, so will use this to avoid recreating the exact same data structures.